FppTest/dp/DpTest.cpp

```diff
@@ -0,0 +1,188 @@
+// ======================================================================
+// \title  DpTest.cpp
+// \author bocchino
+// \brief  cpp file for DpTest component implementation class
+// ======================================================================
+
+#include <cstdio>
+
+#include "FppTest/dp/DpTest.hpp"
+#include "Fw/Types/Assert.hpp"
+
+namespace FppTest {
+
+// ----------------------------------------------------------------------
+// Construction, initialization, and destruction
+// ----------------------------------------------------------------------
+
+DpTest ::DpTest(const char* const compName,
+                U32 u32RecordData,
+                U16 dataRecordData,
+                const U8ArrayRecordData& u8ArrayRecordData,
+                const U32ArrayRecordData& u32ArrayRecordData,
+                const DataArrayRecordData& dataArrayRecordData)
+    : DpTestComponentBase(compName),
+      u32RecordData(u32RecordData),
+      dataRecordData(dataRecordData),
+      u8ArrayRecordData(u8ArrayRecordData),
+      u32ArrayRecordData(u32ArrayRecordData),
+      dataArrayRecordData(dataArrayRecordData),
+      sendTime(Fw::ZERO_TIME) {}
+
+void DpTest ::init(const NATIVE_INT_TYPE queueDepth, const NATIVE_INT_TYPE instance) {
+    DpTestComponentBase::init(queueDepth, instance);
+}
+
+DpTest ::~DpTest() {}
+
+// ----------------------------------------------------------------------
+// Handler implementations for user-defined typed input ports
+// ----------------------------------------------------------------------
+
+void DpTest::schedIn_handler(const NATIVE_INT_TYPE portNum, NATIVE_UINT_TYPE context) {
+    // Request a buffer for Container 1
+    this->dpRequest_Container1(CONTAINER_1_DATA_SIZE);
+    // Request a buffer for Container 2
+    this->dpRequest_Container2(CONTAINER_2_DATA_SIZE);
+    // Request a buffer for Container 3
+    this->dpRequest_Container3(CONTAINER_3_DATA_SIZE);
+    // Request a buffer for Container 4
+    this->dpRequest_Container4(CONTAINER_4_DATA_SIZE);
+    // Request a buffer for Container 5
+    this->dpRequest_Container5(CONTAINER_5_DATA_SIZE);
+    // Get a buffer for Container 1
+    {
+        DpContainer container;
+        Fw::Success status = this->dpGet_Container1(CONTAINER_1_DATA_SIZE, container);
+        FW_ASSERT(status == Fw::Success::SUCCESS, status);
+        // Check the container
+        this->checkContainer(container, ContainerId::Container1, CONTAINER_1_PACKET_SIZE);
```

Comment:
No `dpSend()?`

Reply:
There is some redundancy in the test cases, because there are two ways to request buffers (sync and async). To simplify the testing logic, I skipped the `dpSend` for the sync case and had the component check the container directly. The async case performs the `dpSend` and checks the container at the other end. This way all the code under test is checked, but we avoid some duplicate test code.